### PR TITLE
Feature/dbt in browser static release

### PR DESCRIPTION
**Pull request by JKarlavige**
## Description & motivation
This PR is branched off of PR #2269, and is intended for the Q4 static release version of this project. This PR excludes the python/pyodide work in 2269, as that work will be implemented at a later time.

This is the initial "static" version of the dbt editor, which provides a visual representation on interacting with the product.

Documentation on using this component is available [here](https://www.notion.so/dbtlabs/dbt-Editor-Component-b8634dd292ab41ab8453ea2f33fd0ddc).
## Preview
This page is a demo and will be deleted before take-live.
https://deploy-preview-2643--docs-getdbt-com.netlify.app/docs/dbowser-demo

## To-do before merge
- [x] Implement new manifest with tags
- [x] Remove demo projects/manifests from `website/static/dbt_projects`
- [x] Seed tables not scrolling once collapsed below sidebar
- [x] Refresh PAT in Git before taking live
- [x] Remove `dbowser-demo.md` from `website/docs/docs`
- [x] Remove `dbowser-demo` from `sidebars.js` on line 3

### CODE REVIEW

**Review by krevitt — CHANGES_REQUESTED** [review]
Thanks @JKarlavige!  A few notes here (captured [in this loom walkthrough](https://www.loom.com/share/2e007234af744766ba2162c06b69658e)) before we merge:

1. The folder opening bug appears mostly fixed, but noticed the marts folder isn't populating / displaying the `fct_` model on click (see screenshot below of the manifest, looks like `step_4` is properly populated for that model in the tags).  image

2. In full screen view, lets split the right pane equally vertically between the code and lineage (currently it's ~65% lineage).  Code snippets can be pretty long and are currently being cut off there.  
3. Let's remove the lineage button throughout - for the seeds its not relevant, and for the code snippets we can show lineage in the bottom pane by default (since there's nothing else we'll show there in this version).  For code snippets we can leave the gray bar as a separator (could make it narrower if we have that styled to be easy to change).

**Comment by JKarlavige** [comment]
Sorting out one more item, feel free to hold off on the review until then!

**Comment by JKarlavige** [comment]
This is now ready for review!

**Review by krevitt — DISMISSED** [review]
Love this!  Looks great.

Should we remove the dbowser-demo.md demo page here (so that it doesn't get indexed in search)?  We should just document how to use the component by calling `<dbtEditor project="dbt_sample_project" step="1" />`.

**Comment by JKarlavige** [comment]
> Love this! Looks great.
> 
> Should we remove the dbowser-demo.md demo page here (so that it doesn't get indexed in search)? We should just document how to use the component by calling `<dbtEditor project="dbt_sample_project" step="1" />`.

Yes for sure, i'll be removing any demo pages/content before taking it live. And also have documentation on using the component [here](https://www.notion.so/dbtlabs/dbt-Editor-Component-b8634dd292ab41ab8453ea2f33fd0ddc)

File: website/docs/docs/dbowser-demo.md
Comment (krevitt) [inline website/docs/docs/dbowser-demo.md]: Let's rename `step` to `tag` here (and read the full tag name, ie `step_1`) for universality

Comment (JKarlavige) [reply website/docs/docs/dbowser-demo.md]: Updated prop to `tag`. Think it's worth adding support for multiple tags within this prop?

**Comment by lfracalossiFA** [comment]
Hi Jason.
Help me to understand this bit in your PR text:

> This PR excludes the python/pyodide work in 2269, as that work will be implemented at a later time.

If 2269 is in the commit history, won't it go live when this is merged, regardless? Or does this PR revert that work?

**Comment by JKarlavige** [comment]
> Hi Jason. Help me to understand this bit in your PR text:
> 
> > This PR excludes the python/pyodide work in 2269, as that work will be implemented at a later time.
> 
> If 2269 is in the commit history, won't it go live when this is merged, regardless? Or does this PR revert that work?

This PR removes all pyodide and python work

**Comment by lfracalossiFA** [comment]
I mentioned this when we chatted, but putting it here lest either of us forget: if you click on any of the icons next to the seeds, the IDE disappears and you get an error: "Unable to load data editor."

File: website/plugins/customWebpackConfig/index.js
Comment (john-rock) [inline website/plugins/customWebpackConfig/index.js:30]: Is this an artifact of https://github.com/dbt-labs/docs.getdbt.com/pull/2269?

File: website/.npmrc
Comment (lfracalossiFA) [inline website/.npmrc:2]: For my own edification: is this token necessary so that we can bring in a private package? (I assume react-dbt-dag is private).

File: website/sidebars.js
Comment (lfracalossiFA) [inline website/sidebars.js]: Should we remove this? Or is the plan to remove this before merging, like the page itself?

File: website/plugins/customWebpackConfig/index.js
Comment (lfracalossiFA) [inline website/plugins/customWebpackConfig/index.js:28]: Is this basically making sure it loads Python files as raw data? Is this necessary in this PR if we're not actually loading in any Python?

**Review by lfracalossiFA — DISMISSED** [review]
Nothing too concerning in my review comments; only the note about the "unable to load data editor" error message. Mostly I'm just asking questions for my own edification.

One note that I couldn't comment on directly: what is the purpose of that empty .gitkeep file?

File: website/docs/docs/dbowser-demo.md
Comment (john-rock) [inline website/docs/docs/dbowser-demo.md]: Really clean component implementation.

**Review by john-rock — CHANGES_REQUESTED** [review]
This came out awesome, great work. I know mobile probably isn't the best use case for this but it looks and functions as expected on my iPhone!

One blocker regarding the last level icons (actually noticed @lfracalossiFA already called that out) in the file browser and then a comment for my own clarification.

File: website/plugins/customWebpackConfig/index.js
Comment (JKarlavige) [reply website/plugins/customWebpackConfig/index.js:30]: Yeah it's leftover from the original PR. I left it as it seems to be required for working with any python files on the site, so could help in the rare chance someone else adds a python file to the site. 

I figured it wouldn't hurt to keep, but if there's any concerns with leaving it i'm definitely open to removing it.

File: website/sidebars.js
Comment (JKarlavige) [reply website/sidebars.js]: Yes this will be removed before take-live, thanks for noting this though so it doesn't slip through the cracks.

File: website/.npmrc
Comment (JKarlavige) [reply website/.npmrc:2]: Yes that is correct!

**Comment by JKarlavige** [comment]
> Nothing too concerning in my review comments; only the note about the "unable to load data editor" error message. Mostly I'm just asking questions for my own edification.
> 
> One note that I couldn't comment on directly: what is the purpose of that empty .gitkeep file?

The `.gitkeep` file is used for empty directories. For example, if the `seeds` directory was empty, the .gitkeep would allow the empty directory to still be committed.

I'm not sure how it was added in this case. I downloaded the csv files and dragged them into the repo, so it might have added it automatically. I don't see a need for it in this case, as the seeds directory will always have content in it, otherwise we would just remove the `seeds` directory entirely for that project.

I have checks setup, so if CSV data is missing here is the error message returned. I will remove the gitkeep for this.
image

**Comment by JKarlavige** [comment]
@lfracalossiFA @john-rock The issue with clicking the icon in the sidebar should be fixed!

**Review by john-rock — DISMISSED** [review]
lgtm!

**Comment by JKarlavige** [comment]
Will remove the demo content and will take this live! @krevitt I have documentation available [here](https://www.notion.so/dbtlabs/dbt-Editor-Component-b8634dd292ab41ab8453ea2f33fd0ddc) for when it's time to add this to a page

**Review by krevitt — APPROVED** [review]
looks great!  nice work on this Jason

**Pull request merged**
